File: hoyomi_bridge/lib/core_services/comic/interfaces/comic_modes.dart
class ComicModes {
  static const String leftToRight = 'leftToRight';
  static const String rightToLeft = 'rightToLeft';
  static const String topToBottom = 'topToBottom';
  static const String webToon = 'webToon';

  static const List<String> values = [
    leftToRight,
    rightToLeft,
    topToBottom,
    webToon,
  ];

  static bool isValid(String? value) => values.contains(value);
}

File: lib/core_services/comic/interfaces/comic_modes.ts
export enum ComicModes {
  leftToRight = "leftToRight",
  rightToLeft = "rightToLeft",
  topToBottom = "topToBottom",
  webToon = "webToon"
}

export function modeUsingPageView(mode: ComicModes): boolean {
  return (
    mode === ComicModes.leftToRight ||
    mode === ComicModes.rightToLeft ||
    mode === ComicModes.topToBottom
  )
}
